smithy-cli/src/main/java/software/amazon/smithy/cli/commands/InitCommand.java

```diff
@@ -0,0 +1,170 @@
+
+        // Use templateName if directory is not specified
+        if (directory == null) {
+            directory = template;
+        }
+
+        // Check Git is installed
+        exec(ListUtils.of("git", "clone", "--filter=blob:none", "--no-checkout", "--depth", "1", "--sparse",
+            repositoryUrl, temp.toString()), root);
+
+        // Download template json file
+        exec(ListUtils.of("git", "sparse-checkout", "set", "--no-cone", SMITHY_TEMPLATE_JSON), temp);
+
+        exec(ListUtils.of("git", "checkout"), temp);
+
+        // Retrieve template path from smithy-templates.json
+        String templatePath = readJsonFileAsNode(Paths.get(temp.toString(), SMITHY_TEMPLATE_JSON))
+            .expectObjectNode()
+            .expectMember("templates", String.format(
+                    "Missing expected member `templates` from %s", SMITHY_TEMPLATE_JSON))
+            .expectObjectNode()
+            .expectMember(template, String.format("Missing expected member `%s` from `templates` object", template))
+            .expectObjectNode()
+            .expectMember("path", String.format("Missing expected member `path` from `%s` object", template))
+            .expectStringNode()
+            .getValue();
+
+        // Specify the subdirectory to download
+        exec(ListUtils.of("git", "sparse-checkout", "set", "--no-cone", templatePath), temp);
+
+        exec(ListUtils.of("git", "checkout"), temp);
+
+        IoUtils.copyDir(Paths.get(temp.toString(), templatePath), Paths.get(directory));
+
+        try (ColorBuffer buffer = ColorBuffer.of(env.colors(), env.stderr())) {
+            buffer.println(String.format("Smithy project created in directory: %s", directory), ColorTheme.SUCCESS);
+        }
+    }
+
+    private static String exec(List<String> args, Path directory) {
+        StringBuilder output = new StringBuilder();
+        int code = IoUtils.runCommand(args, directory, output, Collections.emptyMap());
+        if (code != 0) {
+            String errorPrefix = "Unable to run `" + String.join(" ", args) + "`";
+            throw new CliError(errorPrefix + ": " + output);
+        }
+        return output.toString();
+    }
+
+    private Node readJsonFileAsNode(Path jsonFilePath) {
+        return Node.parse(IoUtils.readUtf8File(jsonFilePath));
+    }
+
+    private static final class Options implements ArgumentReceiver {
+        private String template;
+        private String directory;
+        private String repositoryUrl = DEFAULT_REPOSITORY_URL;
+
+        @Override
+        public Consumer<String> testParameter(String name) {
+            switch (name) {
+                case "--template":
+                case "-t":
+                    return value -> template = value;
+                case "--url":
+                case "-u":
+                    return value -> repositoryUrl = value;
+                case "--directory":
+                case "-d":
```

Comment:
We already use `-d` for model discovery. Can you maybe use `--output` and `-o`? This will better match the already in use `--output` parameter used in `smithy build`.